packages/ai-jsx/src/core/errors.ts

```diff
@@ -0,0 +1,10 @@
+export class HttpError extends Error {
```

Comment:
To your point about normalization / keeping core thin – what if we didn't have any of these fields, and just included the Axios response object? 

The way I see it, we have two use-cases:
1. Programmatic log observers wants as much data as possible
2. Humans reading logs in the command line or in the logs wants focused results.

This approach hits a middle ground that doesn't fully satisfy either use-case. 

I think we want less normalization within AI.JSX – so just attach the Axios object. But at the very edge, where we format for the dev, we have an opinionated formatter. (Maybe there's an `OpenAIError` class that contains that opinionation, via a `toString` or `formatForLogs` method or something.)

For the log file, maybe what we want is to log twice:

```
log.trace(error, 'error details');
log.error(summarize(error), 'error summary');
```

I don't love that solution tbh. Another solution (which presumes pino) is to include a `pino-pretty` command that filters which log fields to show, but I don't love that either.

~~~

Having said all that, what we have here is basically fine, and it's probably the case that in the majority of the time, all you need are the fields you've picked out here, and it is genuinely useful to have them normalized across whatever HTTP libraries the various model providers use.

Reply:
TBH because the Axios object is such a dumpster fire when logged I went out of my way to _avoid_ including it in the (logged) exception.

I agree that I'm sort of guessing at a middle ground which is a dicey proposition. I think if I were to make a change here I'd just merge `HttpError` and `OpenAIError`, but I'm going to roll with this for the time being while internally considering `HttpError` a tenuous abstraction that should be revisited if/when we add more service-backed components :)